src/main/java/net/snowflake/ingest/streaming/internal/ConfigureCallHandler.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+package net.snowflake.ingest.streaming.internal;
+
+import static net.snowflake.ingest.streaming.internal.StreamingIngestUtils.executeWithRetries;
+import static net.snowflake.ingest.utils.Constants.RESPONSE_SUCCESS;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import net.snowflake.client.jdbc.internal.apache.http.impl.client.CloseableHttpClient;
+import net.snowflake.ingest.connection.IngestResponseException;
+import net.snowflake.ingest.connection.RequestBuilder;
+import net.snowflake.ingest.connection.ServiceResponseHandler;
+import net.snowflake.ingest.utils.ErrorCode;
+import net.snowflake.ingest.utils.SFException;
+import net.snowflake.ingest.utils.Utils;
+
+/** A class that is used to configure a Snowflake Ingest Storage */
+class ConfigureCallHandler {
+
+  // Object mapper for creating payload, ignore null fields
+  private static final ObjectMapper mapper =
+      new ObjectMapper().setSerializationInclusion(JsonInclude.Include.NON_NULL);
+
+  private final CloseableHttpClient httpClient;
+  private final RequestBuilder requestBuilder;
+  private final ServiceResponseHandler.ApiName apiName;
+  private final String configureEndpoint;
+  private final String role;
+  private final String database;
+  private final String schema;
+  private final String table;
+
+  /**
+   * Builder method to create a {@link ConfigureCallHandler}
+   *
+   * @param httpClient the HTTP client
+   * @param requestBuilder the request builder
+   * @param apiName the API name, used for logging
+   * @param configureEndpoint the configure endpoint
+   * @return a {@link ConfigureCallHandlerBuilder} object
+   */
+  static ConfigureCallHandlerBuilder builder(
+      CloseableHttpClient httpClient,
+      RequestBuilder requestBuilder,
+      ServiceResponseHandler.ApiName apiName,
+      String configureEndpoint) {
+    return new ConfigureCallHandlerBuilder(httpClient, requestBuilder, apiName, configureEndpoint);
+  }
+
+  /** Builder class to build a {@link ConfigureCallHandler} */
+  static class ConfigureCallHandlerBuilder {
```

Comment:
A builder pattern is typically used when the object you're trying to build has many many optional arguments at construction time, which results in a lot of constructor signature combinations.
In this case all the arguments are mandatory, so you can just have one ConfigureCallHandler (public) constructor and take all arguments over there, instead of using a builder.

Reply:
The `role`, `table`, `schema`, and `database` are optional ([ref1](https://github.com/snowflakedb/snowflake-ingest-java/blob/44cc28e77b49c60c5d772a25fe00271bb675670e/src/main/java/net/snowflake/ingest/streaming/internal/ExternalVolumeManager.java#L78C8-L88C26), [ref2](https://github.com/snowflakedb/snowflake-ingest-java/blob/44cc28e77b49c60c5d772a25fe00271bb675670e/src/main/java/net/snowflake/ingest/streaming/internal/InternalStageManager.java#L42C9-L49C22)), should we keep this?